added_mass/proposal.md

```diff
@@ -0,0 +1,255 @@
+* $\mu_{63} = \mu_{36}$ : added mass moment about the Z axis due to linear acceleration in the Z direction, and vice-versa
+* $\mu_{64} = \mu_{46}$ : added mass moment about the Z axis due to angular acceleration about the X direction, and vice-versa
+* $\mu_{65} =\mu_{56}$ : added mass moment about the Z axis due to angular acceleration about the X direction, and vice-versa
+* $\mu_{66}$ : added mass moment about the Z axis due to angular acceleration about the Z direction
+
+&nbsp;
+
+&nbsp;
+
+## Proposed changes
+
+The [`//link/inertial`](http://sdformat.org/spec?ver=1.9&elem=link#link_inertial) term
+in SDF currenly accepts elements to describe the m matrix (mass and moments of inertia),
+as well as a center of mass pose. This proposal suggests exposing each of the 21 elements
+for the mass matrix in addition to that. All added mass elements are optional and default
+to zero if unset. This preserves the behaviour for links that don't have those terms.
+
+### XML spec
+
+A new `<added_mass>` element will be added under `//link/inertial/`. It will contain each of the
+21 matrix elements started by `n` and followed by its index.
+
+```xml
+<inertial>
+  <mass>10</mass>
+  <pose>1 1 1 0 0 0</pose>
+  <inertia>
+    <ixx>0.16666</ixx>
+    <ixy>0</ixy>
+    <ixz>0</ixz>
+    <iyy>0.16666</iyy>
+    <iyz>0</iyz>
+    <izz>0.16666</izz>
+  </inertia>
+  <added_mass>
+    <mu11>1</mu11>
+    <mu12>0</mu12>
+    <mu13>0</mu13>
+    <mu14>0</mu14>
+    <mu15>0</mu15>
+    <mu16>0</mu16>
+    <mu22>1</mu22>
+    <mu23>0</mu23>
+    <mu24>0</mu24>
+    <mu25>0</mu25>
+    <mu26>0</mu26>
+    <mu33>1</mu33>
+    <mu34>0</mu34>
+    <mu35>0</mu35>
+    <mu36>0</mu36>
+    <mu44>0.1</mu44>
+    <mu45>0</mu45>
+    <mu46>0</mu46>
+    <mu55>0.1</mu55>
+    <mu56>0</mu56>
+    <mu66>0.1</mu66>
+  </added_mass>
+</inertial>
+```
+
+### C++ API
+
+The `//link/inertial` element maps to a
+[gz::math::Inertiald](https://gazebosim.org/api/math/6.10/classignition_1_1math_1_1Inertial.html)
+object. To support added mass, the following member functions will be added to the
+`Inertial` class:
+
+```cpp
+/// \brief Constructs an inertial object from the mass matrix for a body
+/// B, about its center of mass Bcm, and expressed in a frame that we’ll
+/// call the "inertial" frame Bi, i.e. the frame in which the components
+/// of the mass matrix are specified (see this class’s documentation for
+/// details). The pose object specifies the pose X_FBi of the inertial
+/// frame Bi in the frame F of this inertial object
+/// (see class’s documentation). The added mass matrix is also expressed
+/// in frame Bi.
+/// \param[in] _massMatrix Mass and inertia matrix.
+/// \param[in] _pose Pose of center of mass reference frame.
+public: Inertial(const MassMatrix3<T> &_massMatrix,
+                 const Pose3<T> &_pose,
+                 const Matrix6<T> &_addedMass)
```

Comment:
I posted an alternative to `Matrix6` in https://github.com/gazebosim/gz-math/pull/442#issuecomment-1159427804

Reply:
I'll take a stab at storing the elements as a `std::array<double, 21>` instead of a `Matrix6d`, mimicking the idea from how `MassMatrix3`stores the moments in vectors instead of a matrix. My main concern is that this may result in a very clunky API.